### PR TITLE
#15: adds basic test coverage

### LINKED ISSUE #15: Constantly measure code coverage
**Is your feature request related to a problem? Please describe.**
Code coverage is currently not measured. So we are a bit blind on this eye. 

**Describe the solution you'd like**
We want to constantly measure code coverage with every pull request and show the current coverage in the README.

### PR DESCRIPTION
**Pull request by timvancleef**
# Pull Request Template

## Description

This PR adds test coverage report to the `pre-push` hook as well as to our github workflow.

The test job will fail if the coverage is below the threshhold defined in `jest.config.js` - currently it is set to 0 🙈 

**Help wanted regarding showing the badge in the readme**
Regarding the integration of a test coverage badge i didnt find any simple solution besides paying for a third party service like https://about.codecov.io/ or creating the badges in a custom github action and commiting the badge SVGs to this repo in order to have a reference for the readme.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Coverage report is saved to `./coverage/lcov.info` as well as printed to the console:

```
---------------|----------|----------|----------|----------|-------------------|
File           |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------|----------|----------|----------|----------|-------------------|
All files      |    33.33 |      100 |    42.86 |    33.33 |                   |
 TestPools.vue |    33.33 |      100 |    42.86 |    33.33 |... 06,109,110,111 |
---------------|----------|----------|----------|----------|-------------------|
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by cremich** [comment]
@timvancleef awesome! Codecov should be free for public repositories, isn't it?

I will check it out later and configure it if it is free.

**Comment by timvancleef** [comment]
> 
> 
> @timvancleef awesome! Codecov should be free for public repositories, isn't it?
> 
> I will check it out later and configure it if it is free.

Oh, of course you are right... this is an open source project 🙈 my bad.
Yup, than we should be fine using codecov.io

**Comment by timvancleef** [comment]
@cremich i added the codecov npm package and an npm task to upload the coverage: `npm run codecov`
Currently the `pull-request` workflow only dumps the coverage report by calling `npm run codecov -- --dump`. To actually upload the coverage report it should be enough to remove `-- --dump`.

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -31,4 +31,6 @@ jobs:
       - name: Check code style
         run: npm run lint
       - name: Test
-        run: npm test
+        run: npm run test:coverage
+      - name: Upload coverage
+        run: npm run codecov -- --dump
```
Comment (timvancleef) [inline .github/workflows/pull-request.yaml:36]: Remove `-- --dump` to actually upload the coverage report.

```suggestion
        run: npm run codecov
```

**Review by cremich — APPROVED** [review]

**Pull request merged**